app/controllers/users_controller.rb

```diff
@@ -111,11 +111,15 @@ def show(for_card: false)
     @user =
       fetch_user_from_params(
         include_inactive:
-          current_user.try(:staff?) || (current_user && SiteSetting.show_inactive_accounts),
+          current_user && (for_card || current_user.staff? || SiteSetting.show_inactive_accounts),
       )
 
     user_serializer = nil
-    if guardian.can_see_profile?(@user)
+    if !current_user&.staff? && !@user.active?
```

Comment:
THis means that staff users will see the full user card for inactive users? I guess previously they were also seeing the full user card for inactive users?

Reply:
Yes, staff users can see full user card as before. I believe we can improve this because there is no indication right now that the user is inactive.